sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala

```diff
@@ -124,4 +135,24 @@ object DateFormatter {
   def apply(zoneId: ZoneId): DateFormatter = {
     getFormatter(None, zoneId)
   }
+
+  def hasDiffResult(s: String, format: String, zoneId: ZoneId): Boolean = {
+    // Only check whether we will get different results between legacy format and new format, while
+    // legacy time parser policy set to EXCEPTION. For legacy parser, DateTimeParseException will
+    // not be thrown. On the contrary, if the legacy policy set to CORRECTED,
+    // DateTimeParseException will address by the caller side.
+    if (LegacyBehaviorPolicy.withName(
+        SQLConf.get.getConf(SQLConf.LEGACY_TIME_PARSER_POLICY)) == EXCEPTION) {
+      val formatter = new LegacySimpleTimestampFormatter(
+        format, zoneId, defaultLocale, lenient = false)
```

Comment:
different places can create different legacy formatter. It's better to always create the legacy formatter, and the new formatter just carry the instance of legacy formatter.

Reply:
Thanks, done in 47d2f80